packages/sdk/src/clients/bridgePool/client.ts

```diff
@@ -0,0 +1,167 @@
+      const { token, amount, lpTokensMinted, liquidityProvider } = typedEvent.args;
+      const tokens = Balances(state.tokens[token] || {});
+      const lpTokens = Balances(state.lpTokens[token] || {});
+      tokens.add(liquidityProvider, amount.toString());
+      lpTokens.add(liquidityProvider, lpTokensMinted.toString());
+      return {
+        ...state,
+        tokens: {
+          ...state.tokens,
+          [token]: tokens.balances,
+        },
+        lpTokens: {
+          ...state.lpTokens,
+          [token]: lpTokens.balances,
+        },
+      };
+    }
+    // event LiquidityRemoved(address indexed token, uint256 amount, uint256 lpTokensBurnt, address liquidityProvider);
+    case "LiquidityRemoved": {
+      const typedEvent = event as LiquidityRemoved;
+      const { token, amount, lpTokensBurnt, liquidityProvider } = typedEvent.args;
+      const tokens = Balances(state.tokens[token] || {});
+      const lpTokens = Balances(state.lpTokens[token] || {});
+      tokens.sub(liquidityProvider, amount.toString());
+      lpTokens.sub(liquidityProvider, lpTokensBurnt.toString());
+      return {
+        ...state,
+        tokens: {
+          ...state.tokens,
+          [token]: tokens.balances,
+        },
+        lpTokens: {
+          ...state.lpTokens,
+          [token]: lpTokens.balances,
+        },
+      };
+    }
+    // event DepositRelayed( bytes32 indexed depositHash, DepositData depositData, address l1Token, RelayData relay, bytes32 relayAncillaryDataHash);
+    case "DepositRelayed": {
+      const typedEvent = event as DepositRelayed;
+      const { depositHash, depositData, relay, relayAncillaryDataHash } = typedEvent.args;
+      return {
+        ...state,
+        relays: {
+          ...state.relays,
+          [relayAncillaryDataHash]: relay,
+        },
+        deposits: {
+          ...state.deposits,
+          [depositHash]: depositData,
+        },
+      };
+    }
+    // event RelaySpedUp(bytes32 indexed depositHash, address indexed instantRelayer, RelayData relay);
+    case "RelaySpedUp": {
+      const typedEvent = event as RelaySpedUp;
+      const { depositHash, instantRelayer } = typedEvent.args;
+      return {
+        ...state,
+        instantRelays: {
+          ...(state.instantRelays || {}),
+          [depositHash]: instantRelayer,
+        },
+      };
+    }
+    // event RelayDisputed(bytes32 indexed depositHash, bytes32 indexed relayHash, address indexed disputer);
+    case "RelayDisputed": {
+      // TODO: finish implementation
+      // const typedEvent = event as RelayDisputed;
+      // const { depositHash, relayHash, disputer } = typedEvent.args;
+      return state;
+    }
+    // event RelayCanceled(bytes32 indexed depositHash, bytes32 indexed relayHash, address indexed disputer);
+    case "RelayCanceled": {
+      // TODO: finish implementation
+      // const typedEvent = event as RelayCanceled;
+      // const { depositHash, relayHash, disputer } = typedEvent.args;
```

Comment:
unsure how to signify relay canceled in state

Reply:
Same q as above.